resourcet: Support exceptions-0.10

ocharles opened the pull request:
Fixes #365

I made the decision to act like a reader monad transformer here, but that might not be correct. Maybe you want to roll back changes to the `IORef` - I'll have to defer to someone who understands this stuff more.

ocharles commented:
Ok, let me see what I can do.

snoyberg commented:
Actually, I should be able to patch this up.

snoyberg commented:
Thanks for the PR, I've added back 0.8 support (skipping 0.9, since it's deprecated), added an upper bound, and added a test case for this kind of bug.

ocharles commented:
Thanks!

On Tue, 10 Apr 2018, 8:36 am Michael Snoyman, <notifications@github.com>
wrote:

> Thanks for the PR, I've added back 0.8 support (skipping 0.9, since it's
> deprecated), added an upper bound, and added a test case for this kind of
> bug.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/snoyberg/conduit/pull/366#issuecomment-380004244>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABRjnxMrauzTQ5dAQQpkRkV23s69j6Uks5tnGENgaJpZM4TMzP1>
> .
>
